src/services/AccomodationService.js

```diff
@@ -0,0 +1,94 @@
+import sequelize from 'sequelize';
+import { ratings } from '../database/models';
+import CommonQueries from './CommonQueries';
+
+/**
+ * @exports
+ * @class CommentService
+ */
+class AccomodationService {
+  /**
+ * Get user rating
+ * @static
+ * @description POST /api/accomodation/ratings
+ * @param {object} req request object
+ * @memberof AccommodationController
+ * @returns {object} data
+ */
+  static async getUserRating(req) {
+    const { id: userId } = req.user;
+    const { accommodationId } = req.body;
+    const userRatingObject = {
+      where: {
+        userId,
+        accommodationId,
+      },
+    };
+    const availableUserRating = CommonQueries.findAll(ratings, userRatingObject);
+    return availableUserRating;
+  }
+
+  /**
+ * users can view total sum rating of that accommodation centre
+ * @static
+ * @memberof AccommodationController
+ * @param {object} req request object
+ * @returns {object} data
+ */
+  static async getRatingSum(req) {
+    const totalRatings = await ratings.findOne({
+      where: { accommodationId: req.body.accommodationId },
+      attributes: ['accommodationId', [sequelize.fn('sum', sequelize.col('rating')), 'totalRating'],
+      ],
+      group: ['accommodationId']
+    });
+    return totalRatings;
+  }
+
+
+  /**
+ * users can view the rating average of that accommodation centre
+ * @static
+ * @description POST /api/accomodation/ratings
+ * @param {object} req request object
+ * @memberof AccommodationController
+ * @returns {object} data
+ */
+  static async getRatingAverage(req) {
+    const averageRating = await ratings.findOne({
+      where: { accommodationId: req.body.accommodationId },
+      attributes: ['accommodationId', [sequelize.fn('AVG', sequelize.col('rating')), 'averageRating'],
+      ],
+      group: ['accommodationId']
+    });
+    return averageRating;
+  }
+
+  /**
+ * users can rate an accommodation centre
+ * @static
+ * @description POST /api/accomodation/ratings
+ * @param {object} req request object
+ * @memberof AccommodationController
+ * @returns {object} data
+ */
+  static async createNewRating(req) {
+    const { rating, review, accommodationId } = req.body;
+    const { id: userId } = req.user;
+    const ratingQueryObject = {
+      userId, rating, review, accommodationId,
+    };
+    const totalRating = await AccomodationService.getRatingSum(req);
+    const averageRating = await AccomodationService.getRatingAverage(req);
+    const newRating = await CommonQueries.create(ratings, ratingQueryObject).then(
+      createdRating => {
+        createdRating.dataValues.totalRatings = totalRating;
+        createdRating.dataValues.averageRating = averageRating;
+        return createdRating;
+      }
+    );
```

Comment:
@ivymwende can we round off average rating to a specific number of decimal parts?
We can maybe set the average rating to have only 1 decimal part
<img width="593" alt="Screen Shot 2019-12-05 at 15 06 27" src="https://user-images.githubusercontent.com/35301129/70238013-0ae8e580-1771-11ea-9521-ab39092cf2f8.png">

Reply:
Oh, I'm afraid that you forgot to first check if the accommodation id exist before rating
<img width="788" alt="Screen Shot 2019-12-05 at 15 11 13" src="https://user-images.githubusercontent.com/35301129/70238641-50f27900-1772-11ea-98d5-7c421388420b.png">